lib/std/json.zig

```diff
@@ -1905,6 +1910,36 @@ fn parseInternal(
                 else => @compileError("Unable to parse into type '" ++ @typeName(T) ++ "'"),
             }
         },
+        .Vector => |vectorInfo| {
+            switch (token) {
+                .ArrayBegin => {
+                    var vec = std.meta.Vector(vectorInfo.len, vectorInfo.child){};
+                    var idx: usize = 0;
+                    errdefer while (idx > 0) {
+                        idx -= 1;
+                        if (idx < vectorInfo.len) {
+                            parseFree(vectorInfo.child, vec[idx], options);
+                        }
+                    };
+                    while (true) : (idx += 1) {
+                        const tok = (try tokens.next()) orelse return error.UnexpectedEndOfJson;
+                        switch (tok) {
+                            .ArrayEnd => break,
+                            else => {},
+                        }
+
+                        if (idx < vectorInfo.len) {
+                            vec[idx] = try parseInternal(vectorInfo.child, tok, tokens, options);
+                        }
```

Comment:
```suggestion
                        if (idx >= vectorInfo.len) return error.VectorLengthMismatch;
                        
                        vec[idx] = try parseInternal(vectorInfo.child, tok, tokens, options);
```
This prevents doing un-necessary work parsing array elements that would never fit in the vector